reactDashboard/src/components/DataCharts/DataChartsUtils.test.js

```diff
@@ -104,3 +203,104 @@ describe("calculateDateRange", () => {
     expect(result).toStrictEqual(EMPTY_DATE_RANGE);
   });
 });
+
+describe("getChartYMax", () => {
+  const TEST_DATE_RANGE = {
+    startDate: Date.parse("2020-01-01"),
+    endDate: Date.parse("2020-01-08"),
+  };
+  const EMPTY_DATE_RANGE = {
+    startDate: Date.parse("1980-01-01"),
+    endDate: Date.parse("1980-01-08"),
+  };
+
+  const TEST_EMPTY_DATASET = [
+    {
+      data: [],
+    },
+  ];
+  const TEST_SINGLE_DATASET = [
+    {
+      data: TEST_MAX_DATA_1,
+    },
+  ];
+  const TEST_DOUBLE_DATASET = [
+    {
+      data: TEST_MAX_DATA_1,
+    },
+    {
+      data: TEST_MAX_DATA_2,
+    },
+  ];
+
+  it("datasets empty", () => {
+    expect(getChartYMax([], TEST_DATE_RANGE)).toStrictEqual(0);
+  });
+  it("datasets null", () => {
+    expect(getChartYMax(null, TEST_DATE_RANGE)).toStrictEqual(0);
+  });
+  it("datasets undefined", () => {
+    expect(getChartYMax(undefined, TEST_DATE_RANGE)).toStrictEqual(0);
+  });
+  it("no data in dateRange", () => {
+    expect(getChartYMax(TEST_SINGLE_DATASET, EMPTY_DATE_RANGE)).toStrictEqual(
+      0
+    );
+  });
+  it("dateRange null", () => {
+    expect(getChartYMax(TEST_SINGLE_DATASET, null)).toStrictEqual(799);
+  });
+  it("dateRange undefined", () => {
+    expect(getChartYMax(TEST_DOUBLE_DATASET)).toStrictEqual(4056);
+  });
+  it("dateRange with double dataset and first array is bigger", () => {
+    expect(
+      getChartYMax(TEST_DOUBLE_DATASET, {
+        startDate: Date.parse("2020-01-03"),
+        endDate: Date.parse("2020-01-04"),
+      })
+    ).toStrictEqual(771);
+  });
+  it("dateRange with double dataset and second array is bigger", () => {
+    expect(
+      getChartYMax(TEST_DOUBLE_DATASET, {
+        startDate: Date.parse("2020-01-06"),
+        endDate: Date.parse("2020-01-07"),
+      })
+    ).toStrictEqual(842);
+  });
+  it("dataset without data points", () => {
+    expect(getChartYMax(TEST_EMPTY_DATASET)).toStrictEqual(0);
+  });
+});
+
+describe("getMaxTicks", () => {
+  const TEST_YMAX_NULL = null;
+  const TEST_YMAX_ZERO = 0;
+  const TEST_YMAX_ONE = 1;
+  const TEST_YMAX_FIVE = 5;
+  const TEST_YMAX_TWENTY = 20;
+  const TEST_YMAX_MILLION = 1000000;
+
+  it("yMax is null", () => {
```

Comment:
Your call, but I generally put calls to test simple pure functions into a single testcase - one it() testcase containing all the expects below